common/src/main/java/com/microsoft/identity/common/internal/eststelemetry/EstsTelemetry.java

```diff
@@ -321,17 +363,63 @@ String getCurrentTelemetryHeaderString() {
             return null;
         }
 
-        return currentTelemetry.getCompleteTelemetryHeaderString();
+        return currentTelemetry.getCompleteHeaderString();
     }
 
-    String getLastTelemetryHeaderString() {
-        RequestTelemetry lastTelemetry = loadLastRequestTelemetryFromCache();
-
-        if (lastTelemetry == null) {
+    private String getLastTelemetryHeaderString() {
+        if (mLastRequestTelemetryCache == null) {
             return null;
         }
 
-        return lastTelemetry.getCompleteTelemetryHeaderString();
+        LastRequestTelemetry lastRequestTelemetry = (LastRequestTelemetry) mLastRequestTelemetryCache.getRequestTelemetryFromCache();
+
+        if (lastRequestTelemetry == null) {
+            // we did not have anything in the telemetry cache for the last request
+            // let's create a new object based on the data available from the current request object
+            // and return the header string formed via that object
+            final String correlationId = DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID);
+            final CurrentRequestTelemetry currentRequestTelemetry = mTelemetryMap.get(correlationId);
+            lastRequestTelemetry = new LastRequestTelemetry(currentRequestTelemetry.getSchemaVersion());
+            lastRequestTelemetry = (LastRequestTelemetry) lastRequestTelemetry.derive(currentRequestTelemetry);
```

Comment:
nit: the term 'derive', along with the comment above, gave me an impression that we're actually copying everything from CurrentRequestTelemetry. 

Should we rename this to something along the line of '_copySharedValues_'?

Reply:
Addressed